### PR TITLE
Feature/split effect table settings

### PR DESCRIPTION
**Pull request by kzalite**

### CODE REVIEW

File: app/js/workspace/workspaceSettingsService.js [hunk 1]
```diff
@@ -98,8 +134,19 @@ define(['angular', 'lodash'], function(angular, _) {
     }
 
     function getDefaults() {
+      var defaultSettings = angular.copy(DEFAULT_SETTINGS);
+      defaultSettings.isRelativeProblem = workspaceSettings.isRelativeProblem;
+      defaultSettings.hasNoEffects = workspaceSettings.hasNoEffects;
+      defaultSettings.hasNoDistributions = workspaceSettings.hasNoDistributions;
```
Comment (DanielReid) [inline app/js/workspace/workspaceSettingsService.js]: _.merge + _.pick

File: app/js/workspace/workspaceSettingsService.js [hunk 2]
```diff
@@ -109,8 +156,35 @@ define(['angular', 'lodash'], function(angular, _) {
...
+
+    function getWarnings(settings) {
+      var warnings = [];
+      if (hasNoEnteredData(settings)) {
+        warnings.push('No entered data available.');
+      } else if (hasNoEnteredEffect(settings)) {
+        warnings.push('No entered data available for deterministic analysis.');
+      } else if (hasNoEnteredDistribution(settings)) {
+        warnings.push('No entered data available for SMAA analysis.');
+      }
```
Comment (DanielReid) [inline app/js/workspace/workspaceSettingsService.js]: if every push is in an else branch you can just return the warning directly

File: app/js/workspace/workspaceSettings.html [hunk 3]
```diff
@@ -20,29 +20,34 @@ <h3>Settings</h3>
       </div>
 
       <div class="cell">
-        Effects table display mode
+        Display mode
```
[the remark this answers is not in the document]
Comment (chinook25) [reply app/js/workspace/workspaceSettings.html]: But it does also influence other places e.g. the overview tab and the scales table. Should these then also be mentioned?

File: app/js/effectsTable/effectsTableCellDirective.html [hunk 4]
```diff
@@ -1,28 +1,32 @@
...
+    </span>
+  </div>
 
-<div ng-if="effectsDisplay === 'smaaDistributions'">
-  {{distributionLabel}}
-  <span ng-if="distributionLabel === ''" tooltip="No values entered">
-    &nbsp;
-  </span>
-</div>
+  <div ng-if="displayMode === 'values'">
```
[the remark this answers is not in the document]
Comment (chinook25) [reply app/js/effectsTable/effectsTableCellDirective.html]: For SMAA yes, for effects sometimes, but mostly taken directly from the problem.

File: app/js/workspace/workspaceSettings.html (hunk 3 above)
[the remark this answers is not in the document]
Comment (DanielReid) [reply app/js/workspace/workspaceSettings.html]: The clearer the better. The main point is that a choice like 'analysis type' is meaningless to a user without context. At a minimum there should be probably be context help again on the choices.

File: app/js/workspace/workspaceSettingsService.js (hunk 2 above)
Comment (chinook25) [reply app/js/workspace/workspaceSettingsService.js]: True, but I wanted it to follow the same style as we do in other places. Also this way more warnings can be added easily if ever needed.

Comment (DanielReid) [reply app/js/workspace/workspaceSettingsService.js]: IMO it's distracting to use this format if it's not necessary, I kept thinking I was missing something :) context-free code!

File: database.pg.sql [hunk 5]
```diff
@@ -159,3 +159,49 @@ COMMIT;
...
+  jsonb_set(settings, '{settings, analysisType}', newSettings.analysisType->'analysisType'), 
+  '{settings, displayMode}', 
+  newSettings.displayMode->'displayMode'
+)
+FROM newSettings 
+WHERE workspacesettings.workspaceId = newSettings.workspaceId;
+
+UPDATE workspacesettings 
+SET settings = settings #-'{settings, effectsDisplay}' 
+WHERE settings#>'{settings, effectsDisplay}' IS NOT NULL;
```
Comment (DanielReid) [inline database.pg.sql]: interested: is the WHERE necessary here?

File: database.pg.sql [hunk 6]
```diff
@@ -159,3 +159,49 @@ COMMIT;
...
+
+COMMIT;
+--rollback UPDATE workspacesettings
+--rollback SET settings='{}'::jsonb
```
Comment (DanielReid) [inline database.pg.sql]: rollback should be possible (if not fun)

File: app/js/workspace/workspaceSettings.html (hunk 3 above)
[the remark this answers is not in the document]
Comment (kzalite) [reply app/js/workspace/workspaceSettings.html]: Changed to 'Measurements display mode' and 'Values used for analysis'

File: database.pg.sql (hunk 5 above)
Comment (kzalite) [reply database.pg.sql]: Removed

**Pull request merged**